server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverConditions.java

```diff
@@ -256,6 +257,60 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
+     * @throws ElasticsearchParseException if the input is invalid, if there are unknown conditions or invalid format values.
+     */
+    public static RolloverConditions parseSetting(String input, String setting) {
+        String[] sConditions = input.split(",");
+        RolloverConditions.Builder builder = newBuilder();
+        for (String sCondition : sConditions) {
+            String[] keyValue = sCondition.split("=");
+            if (keyValue.length != 2) {
+                throw new ElasticsearchParseException("Invalid condition: '{}', format must be 'condition=value'", sCondition);
+            }
+            var condition = keyValue[0];
+            var value = keyValue[1];
+            if (MAX_SIZE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMaxIndexSizeCondition(ByteSizeValue.parseBytesSizeValue(value, setting));
+            } else if (MAX_PRIMARY_SHARD_SIZE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMaxPrimaryShardSizeCondition(ByteSizeValue.parseBytesSizeValue(value, setting));
+            } else if (MAX_AGE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMaxIndexAgeCondition(TimeValue.parseTimeValue(value, setting));
+            } else if (MAX_DOCS_FIELD.getPreferredName().equals(condition)) {
+                builder.addMaxIndexDocsCondition(parseLong(value, setting));
+            } else if (MAX_PRIMARY_SHARD_DOCS_FIELD.getPreferredName().equals(condition)) {
+                builder.addMaxPrimaryShardDocsCondition(parseLong(value, setting));
+            } else if (MIN_SIZE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMinIndexSizeCondition(ByteSizeValue.parseBytesSizeValue(value, setting));
+            } else if (MIN_PRIMARY_SHARD_SIZE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMinPrimaryShardSizeCondition(ByteSizeValue.parseBytesSizeValue(value, setting));
+            } else if (MIN_AGE_FIELD.getPreferredName().equals(condition)) {
+                builder.addMinIndexAgeCondition(TimeValue.parseTimeValue(value, setting));
+            } else if (MIN_DOCS_FIELD.getPreferredName().equals(condition)) {
+                builder.addMinIndexDocsCondition(parseLong(value, setting));
+            } else if (MIN_PRIMARY_SHARD_DOCS_FIELD.getPreferredName().equals(condition)) {
+                builder.addMinPrimaryShardDocsCondition(parseLong(value, condition));
+            } else {
+                throw new ElasticsearchParseException("Unknown condition: '{}'", condition);
+            }
+        }
+        return builder.build();
+    }
+
+    private static Long parseLong(String sValue, String settingName) {
+        try {
+            return Long.parseLong(sValue);
+        } catch (NumberFormatException e) {
+            throw new ElasticsearchParseException(
```

Comment:
Should we throw `SettingsException` here?

Reply:
Hm, yes, I can see that being more appropriate :).